roles/tests-integration/tests/common/sniffer.rs

```diff
@@ -50,17 +52,56 @@ pub struct Sniffer {
     upstream_address: SocketAddr,
     messages_from_downstream: MessagesAggregator,
     messages_from_upstream: MessagesAggregator,
+    interrupt_messages: Vec<InterruptMessage>,
+}
+
+#[derive(Debug, Clone)]
+pub struct InterruptMessage {
+    direction: MessageDirection,
+    expected_message_type: MsgType,
+    response_message: PoolMessages<'static>,
+    response_message_type: MsgType,
+    break_on: bool,
+}
+
+impl InterruptMessage {
```

Comment:
this name feels a bit counterintuitive from the user perspective

"interrupt" implies that things will stop happening and this is the end of the test, which is not necessarily true

two alternative names that come to mind now are:
- `InterceptMessage`
- `ModifyMessage`